python/ray/util/data/interface.py

```diff
@@ -0,0 +1,26 @@
+from typing import Iterable
+
+import pandas as pd
+
+
+class SourceShard:
```

Comment:
Do we need this interface? It seems like `from_source_shards()` is not used.

Reply:
We can also make this internal (add underscore => _SourceShard, _from_source_shards)

Reply:
`SourceShard ` should be useful when we support more data format. I have removed `from_source_shards `.